e2e/container_registry_test.go

```diff
@@ -44,51 +44,35 @@ var _ = Describe("enforcing a Container Registry", func() {
 				Allowed:      []string{"docker.io", "docker.tld"},
 				AllowedRegex: `quay\.\w+`,
 			},
-			NamespacesMetadata: v1alpha1.AdditionalMetadata{},
-			ServicesMetadata:   v1alpha1.AdditionalMetadata{},
-			StorageClasses:     &v1alpha1.StorageClassesSpec{},
-			IngressClasses:     &v1alpha1.IngressClassesSpec{},
-			LimitRanges:        []corev1.LimitRangeSpec{},
-			NodeSelector:       map[string]string{},
-			ResourceQuota:      []corev1.ResourceQuotaSpec{},
 		},
 	}
 	JustBeforeEach(func() {
 		EventuallyCreation(func() error {
-			return k8sClient.Create(context.TODO(), tnt.DeepCopy())
+			tnt.ResourceVersion = ""
+			return k8sClient.Create(context.TODO(), tnt)
 		}).Should(Succeed())
 	})
 	JustAfterEach(func() {
-		Expect(k8sClient.Delete(context.TODO(), tnt.DeepCopy())).Should(Succeed())
+		Expect(k8sClient.Delete(context.TODO(), tnt)).Should(Succeed())
 	})
+
 	It("should add labels to Namespace", func() {
 		ns := NewNamespace("registry-labels")
 		NamespaceCreation(ns, tnt, defaultTimeoutInterval).Should(Succeed())
-
-		Eventually(func() bool {
-			if err := k8sClient.Get(context.Background(), types.NamespacedName{Name: ns.Name}, ns); err != nil {
-				return false
+		Eventually(func() (ok bool) {
+			Expect(k8sClient.Get(context.Background(), types.NamespacedName{Name: ns.Name}, ns)).Should(Succeed())
+			ok, _ = HaveKeyWithValue("capsule.clastix.io/allowed-registries", "docker.io,docker.tld").Match(ns.Annotations)
```

Comment:
Although is an upside-down assertion, this works well since we don't know when the Namespace has been totally processed by the controller.